### DISCUSSION TITLE
MPI Support

### DISCUSSION THREAD

**Discussion by prudhomm** [DavidAce/h5pp#21] [Q&A]
Hi

very nice work on h5cpp, the c++ code looks indeed much more modern ;)

I have a few questions regarding h5cpp and mpi support :

- what is the state of mpi support in h5cpp ?
- can h5cpp be used readily in a parallel environment ?
- Has it been installed on supercomputers with taylored environments (mpi,hdf5...)

thank you very much in advance

**Comment by DavidAce** [comment]
Thank you for your feedback! I'll do my best to answer:

* MPI support in h5pp is largely untested, but some of the h5pp constructs may still work. More specifically, multi-process write to a single file is untested, but h5pp should work fine in an MPI context as long as a single process writes to a single file at any one time. Since MPI support is handled by the underlying HDF5 library, my belief is that h5pp mostly just needs to propagate the "H5P" property lists that are used to set up the MPI communication for HDF5. This is currently done by h5pp. However, I expect that an MPI user would end up using a mixture of the HDF5 C API for MPI-specific tasks and h5pp for high-level tasks. Long term, I would like to improve support MPI support and expose more of the HDF5 C API, so I welcome any sort of feedback that users may have on this.
* If by parallel environment you mean "as part of an MPI program", then yes but only in the sense I described in the point above. 
* It has been installed on supercomputers by myself and many of my colleagues, and I suspect that many others have as well, based on the scientific computing background that h5pp users seem to have.   We use it on a handful of Slurm clusters where the HDF5 library (compiled with or without MPI support) is loaded from the cluster modules or compiled from source---as long as you can link the HDF5 library, or if the CMake find_package(HDF5) call succeeds, it should work.

**Comment by prudhomm** [comment]
thanks @DavidAce actually we are in the case multiple process write to a single file. I will check if that works.
